### DISCUSSION TITLE
Transfer QuerycClient to 3rd part library

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#2033] [Q&A]
Hello,

I have a 3rd party component that renders columns on the DOM from an external JS library.
Each of those coloumns have ob his own a formatter that is transfered a JSK element to render on screen.

How can i transfer the query client to that element so i can consume muation on on clikc event.
Right now im getting query client is undefined
Any elegant way to do so..... can i transfer the entorie obejct to the parent element ref and then use it ?
Thanks!

**Comment by TkDodo** [comment]
> Right now im getting query client is undefined

I do not understand what you are trying to do. That being said, the query cache is basically just an object. You can get it from the queryClient. The queryClient is the class you create with `new QueryClient`. It is also distributed via react context when you create a `QueryClientProvider`. The cache can be retrieved via `queryClient.getQueryCache()`

**Reply by cubecleveland** [reply]
Thanks Tanner, i got it tow kr bu using the queryClient provider as you described.
